### DISCUSSION TITLE
How to enable auto-update for Microsoft, Apple and Android Store published .MAUI Blazor App?

### DISCUSSION THREAD

**Discussion by vsfeedback** [dotnet/maui#24201] [Q&A]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/How-to-enable-auto-update-for-Microsoft/10711489)._

---
We are developing an app which is based on MAUI.NET Blazor Hybrid, would like to force users to install the latest version before they start using the app when an update is available on Microsoft Store, Apple Apps Store, and Android Play Store.

Looking for a possible solution to enforce to use latest version and avoid using old version.

**Comment by jarlem** [comment]
Not the greatest user experience, but if you wanna do it: 
Get the version of the app running on the device and ask the server if it's the latest version. If not, pop message with a a button that open the store where they can upgrade. If they do not upgrade you can "lock" the app and make it "not usable".

**Reply by ewerspej** [reply]
When an API has breaking changes, sometimes apps require an update before they can work properly again. It's quite a good UX actually if the app notifies you about this before it starts crashing or malfunctioning for no apparent reasons. If it then let's you update the app or at least points you to the App Store/Play Store page where the update can be installed (in In-App Updates are not implemented), then I consider that great UX.
